platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -125,6 +125,59 @@ Refer to Gradle’s link:compatibility.html#kotlin[compatibility matrix] for det
 
 NOTE: Plugins written using the Kotlin DSL and published with Gradle 7.x or 8.x remain compatible with Gradle 6.8 and newer.
 
+==== Upgrade to Groovy 4.0.26
+
+Groovy has been updated to https://groovy-lang.org/changelogs/changelog-4.0.26.html[4.0.26], from 3.0.24.
+Also see the https://groovy-lang.org/releasenotes/groovy-4.0.html[4.0 release notes] for more details on the changes from 3.x to 4.x.
+
+===== Properties with `is`-prefix and `Boolean` types are no longer usable by Groovy
+Groovy 4 no longer detects getters with `is`-prefix and `Boolean` type as a property. Gradle's own interpretation of properties currently remains
+the same, but will be changed in Gradle 10.0 to be consistent with Groovy 4. See <<#groovy_boolean_properties, the deprecation about this>> for more details.
+
+===== Non-abstract managed properties may no longer be attached
+**TODO, waiting on https://github.com/gradle/gradle/issues/33215 resolution**
+
+===== Closures may prefer the delegate over the owner in more cases when using the `DELEGATE_FIRST` strategy
+Groovy 4.0.26 has changed the behavior of closures using the `DELEGATE_FIRST` strategy. Dynamic lookups for properties and methods will now prefer the delegate over the owner.
+This can result in different behavior when using closures in Gradle scripts, such as certain methods not being found or (e.g. with `with(...) { }`) map keys being preferred over properties or methods
+from outside the closure.
+
+Workarounds include using `@CompileStatic` to avoid the dynamic lookup, or prefixing the call with `owner.`, `this.` or `super.` as appropriate.
```

Comment:
```suggestion
Workarounds include using `@CompileStatic` to avoid dynamic lookups, or explicitly qualifying calls with `owner.`, `this.`, or `super.` as needed.
```